src/ILVerification/src/AccessVerificationHelpers.cs

```diff
@@ -252,18 +252,21 @@ private static bool CanAccessFamily(TypeDesc currentType, TypeDesc targetTypeDef
 
         private static bool GrantsFriendAccessTo(this ModuleDesc module, ModuleDesc friendModule)
         {
-            var assembly = (EcmaAssembly)module;
-            var friendName = ((IAssemblyDesc)friendModule).GetName();
-
-            foreach (var attribute in assembly.GetDecodedCustomAttributes("System.Runtime.CompilerServices", "InternalsVisibleToAttribute"))
+            var assembly = module as EcmaAssembly;
+            if (assembly != null)
```

Comment:
This will avoid the exception, but it is not quite the right fix. The right fix would be to find the containing assembly for the netmodule and use that for the access check.

Reply:
Thanks for clarification.
So this means, we need  something along the lines of:

```cs
        private static bool GrantsFriendAccessTo(this ModuleDesc module, ModuleDesc friendModule)
        {
            var assembly = module as EcmaAssembly;
            if (assembly == null)
            {
                assembly = GetAssemblyForNetModule(module as EcmaModule);
            }
            . . .
```

Reply:
Correct